extensions/undertow/spi/src/main/java/io/quarkus/undertow/deployment/UndertowStaticResourcesBuildStep.java

```diff
@@ -32,21 +34,26 @@
     protected static final String META_INF_RESOURCES = "META-INF/resources";
 
     @BuildStep
-    void handleGeneratedWebResources(BuildProducer<GeneratedResourceBuildItem> generatedResources,
+    void handleGeneratedWebResources(Capabilities capabilities, BuildProducer<GeneratedResourceBuildItem> generatedResources,
             List<GeneratedWebResourceBuildItem> generatedWebResources) throws Exception {
+        if (!capabilities.isPresent(Capability.SERVLET)) {
```

Comment:
I wonder if we should add an annotation at the class level instead of having conditions in every build step. @stuartwdouglas do you think that would be worth it?

Reply:
Use the `onlyIf` method in the `@BuildStep` annotation?

Reply:
How can I retrieve the capabilities inside a only if?

Reply:
Inject the `Capabilities` in the class' constructor

Reply:
Does not work.
`BooleanSupplierFactoryBuildItem` only supports injection of LaunchMode, DevModeType, and ConfigRoot.
https://github.com/quarkusio/quarkus/blob/main/core/deployment/src/main/java/io/quarkus/deployment/BooleanSupplierFactoryBuildItem.java#L42

The doc also states:
```
If you need to make your build step conditional on the presence or absence of another extension, you can use Capabilities for that.
```
https://quarkus.io/guides/writing-extensions#conditional-step-inclusion
Which has an example, doing the same `if (!capabilities.isPresent(Capability.SERVLET)) {` I am doing right now.

Reply:
I think @gsmet was talking about adding a new feature to make it easier to handle capabilities. I don't think it should hold up this PR, but an annotation based approach would be nicer.